qucs/qucs-help/main.cpp

```diff
@@ -46,74 +47,29 @@ tQucsSettings QucsSettings; // application settings
 // Loads the settings file and stores the settings.
 bool loadSettings()
 {
-  bool result = true;
-
-  QFile file(QDir::homePath()+QDir::convertSeparators ("/.qucs/helprc"));
-  if(!file.open(QIODevice::ReadOnly))
-    result = false; // settings file doesn't exist
-  else {
-    QTextStream stream(&file);
-    QString Line, Setting;
-    while(!stream.atEnd()) {
-      Line = stream.readLine();
-      Setting = Line.section('=',0,0);
-      Line = Line.section('=',1,1);
-      if(Setting == "HelpWindow") {
-	QucsSettings.x  = Line.section(",",0,0).toInt();
-	QucsSettings.y  = Line.section(",",1,1).toInt();
-	QucsSettings.dx = Line.section(",",2,2).toInt();
-	QucsSettings.dy = Line.section(",",3,3).toInt();
-	break; }
-    }
-    file.close();
-  }
-
-  file.setFileName(QDir::homePath()+QDir::convertSeparators ("/.qucs/qucsrc"));
-  if(!file.open(QIODevice::ReadOnly))
-    result = true; // qucs settings not necessary
-  else {
-    QTextStream stream(&file);
-    QString Line, Setting;
-    while(!stream.atEnd()) {
-      Line = stream.readLine();
-      Setting = Line.section('=',0,0);
-      Line = Line.section('=',1,1).trimmed();
-      if(Setting == "Font")
-	QucsSettings.font.fromString(Line);
-      else if(Setting == "Language")
-	QucsSettings.Language = Line;
-    }
-    file.close();
-  }
-  return result;
+  QSettings settings("qucs","qucs");
+  settings.beginGroup("QucsHelp");
+  if(settings.contains("x"))QucsSettings.x=settings.value("x").toInt();
+  if(settings.contains("y"))QucsSettings.y=settings.value("y").toInt();
+  if(settings.contains("dx"))QucsSettings.dx=settings.value("dx").toInt();
+  if(settings.contains("dy"))QucsSettings.dy=settings.value("dy").toInt();
+  settings.endGroup();
+  if(settings.contains("font"))QucsSettings.font.fromString(settings.value("font").toString());
+  if(settings.contains("Language"))QucsSettings.Language=settings.value("Language").toString();
+  return true;
 }
 
 // #########################################################################
 // Saves the settings in the settings file.
 bool saveApplSettings(QucsHelp *qucs)
 {
-  if(qucs->x() == QucsSettings.x)
```

Comment:
> as now the settings are saved even

this seems to be a regression to me. writing out a file unnecessarily is bad practice. this should not be changed, particularly because it was already there.

what's even worse: does `QSettings settings ("qucs","qucs");` *parse* the configuration?

> But the end result should be practically the same

i cant see, how the settings are written out *at all* (is this hidden in the destructor of qucssettings?!). maybe there should be a global handle on settings, with a "flush()" member, instead of random file i/o.

    auto& settings=global_qucssetitings;
    settings.set("x", 17);
    settings.flush(); // write out lazily, and error handling here.

makes sense, no? i don't think that touching the code without actually imeplementing it in a sane way is worth the trouble (review/testing). it won't converge.